### PR TITLE
Fixes #19966: Update pip install script source

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/19966

### CODE REVIEW

File: scripts/rudder-setup/test-ncf.sh [hunk 1]
```diff
@@ -43,11 +43,18 @@ install_test_dependencies() {
     if [ $TESTINFRA -eq 1 ];
     then
       # install pip
+      PIP=pip
```
Comment (peckpeck) [inline scripts/rudder-setup/test-ncf.sh:46]: it would be more readable to put this in the else

File: scripts/rudder-setup/test-ncf.sh [hunk 2]
```diff
@@ -43,11 +43,18 @@ install_test_dependencies() {
...
+      ${PM_INSTALL} python3-pip || /bin/true
+      if command -v pip3 >/dev/null 2>&1
+      then
+        PIP="pip3"
+      else
+        curl https://bootstrap.pypa.io/pip/2.7/get-pip.py -o get-pip.py || /bin/true
+        python get-pip.py || /bin/true
+      fi
+      $PIP install -U six || /bin/true
+      $PIP install -U testinfra --ignore-installed six || /bin/true
```
Comment (peckpeck) [inline scripts/rudder-setup/test-ncf.sh:57]: isn't six mandatory ?
why do you || true everything ?

By the way there is no need to use the full path for true, it's just more verbose

**Comment by Fdall** [comment]
PR updated with a new commit

File: scripts/rudder-setup/test-ncf.sh [hunk 3]
```diff
@@ -44,10 +44,17 @@ install_test_dependencies() {
     then
       # install pip
       ${PM_INSTALL} python
-      curl https://bootstrap.pypa.io/get-pip.py -o get-pip.py || /bin/true
-      python get-pip.py || /bin/true
-      pip install -U six || /bin/true
-      pip install -U testinfra --ignore-installed six || /bin/true
+      ${PM_INSTALL} python3-pip
```
Comment (peckpeck) [inline scripts/rudder-setup/test-ncf.sh:47]: this may not work you probably need the || true here

**Comment by Fdall** [comment]
PR updated with a new commit

**Review by peckpeck — APPROVED** [review]

**Pull request merged**